torch/csrc/autograd/python_variable.cpp

```diff
@@ -2353,24 +2367,107 @@ c10::intrusive_ptr<TensorImpl> ConcretePyInterpreterVTable::detach(
 }
 
 bool ConcretePyInterpreterVTable::is_contiguous(
+    const c10::TensorImpl* self,
+    at::MemoryFormat memory_format) const {
+  pybind11::gil_scoped_acquire gil;
+  at::impl::MaybeSetTLSOnEntryGuard guard;
+
+  py::object out;
+  if (memory_format == at::MemoryFormat::Contiguous) {
+    // For backwards compatibility
+    out = torchDispatchFromTensorImpl(
+        self,
+        "is_contiguous",
+        py::module::import("torch")
+            .attr("ops")
+            .attr("aten")
+            .attr("is_contiguous")
+            .attr("default")
+            .ptr(),
+        "torch.ops.aten");
+  } else {
+    out = torchDispatchFromTensorImpl(
+        self,
+        "is_contiguous",
+        py::module::import("torch")
+            .attr("ops")
+            .attr("aten")
+            .attr("is_contiguous")
+            .attr("memory_format")
+            .ptr(),
+        "torch.ops.aten",
+        {py::cast(memory_format)});
+  }
+
+  if (out.is(py::none())) {
```

Comment:
I keep wondering if we want to migrate all of these to `NotImplemented` instead of `None`?

Reply:
So... I actually had this return NotImplemented originally, but it is not consistent with how the other functions that support None work (so I'd have to change them do, adding MORE changes to a big patch), and also it has a weird interaction with multiple dispatch (since NotImplemented triggers us to check if there are other arguments that have a torch dispatch implementation)

Reply:
Ok!
The interaction with other dispatch sounds especially good to me.
I'll stop worrying about this then.